astropy/io/misc/asdf/tags/transform/tabular.py

```diff
@@ -47,12 +47,12 @@ def from_tree_transform(cls, node, ctx):
     @classmethod
     def to_tree_transform(cls, model, ctx):
         node = {}
-        node["fill_value"] = model.fill_value
+        if model.fill_value is not None:
+            node["fill_value"] = model.fill_value
```

Comment:
I think the above two lines are not necessary because `fill_value=None` has and is an allowed value but is not the default value.
The `name` attribute is a bug.

Reply:
If we want to write an explicit `fill_value: null` to the ASDF file, then we'll need to update the schema:

https://github.com/asdf-format/asdf-standard/blob/master/schemas/stsci.edu/asdf/transform/tabular-1.2.0.yaml#L50

That will need to become something like:

```
fill_value:
  anyOf:
    - type: number
    - type: "null"
```

Reply:
Ah, right. So let's do it. 
And we'll keep the current PR as is. I'll only move the milestone to 4.0.2 as it can go in the LTS release.

Reply:
You'll have to move the change log to 4.0.2 but feel free to merge after that.